crates/loader/src/map.rs

```diff
@@ -48,6 +51,7 @@ fn spawn_map(
     task: Option<ResMut<MapLoadingTask>>,
     mut move_focus_events: EventWriter<MoveFocusEvent>,
     game_config: Res<GameConfig>,
+    entities_to_remove: Query<Entity, Or<(With<Handle<Scene>>, With<RemoveBeforeLoad>)>>,
```

Comment:
we might want a better strategy for removing stuff between loading states, I would err on the side of tagging everything that should not be removed (as I guess there should be a less amount of entities) rather than the opposite, but that was easier to implement.

I might have missed edge cases like the selection rectangle, or projectiles ?

Reply:
I think that the desired solution to this problem is to add on exist system(s) to every plugin so that the responsibility is moved to where the entities / resources were inserted in the first place.

This all thing might not be needed to add map selection (see the other comment): I would just implement simple main menu where you click on a button with a map name to start the game. You would have to restart the game to select another map to decouple proper system clenaups with map selection.

Reply:
Related #135.